### ISSUE TITLE
Incorrect warning about 

### ISSUE THREAD

**Issue by nomennescio** [llvm/llvm-project#57552]
This code
```
```
generates this warning on Clang8/C18:
```
solution.c:14:25: warning: pointer/integer type mismatch in conditional expression ('void *' and 'int') [-Wconditional-type-mismatch]
void *x(void*y){return y?y:(1,0);}
                        ^~ ~~~~~
```

However, the 0 in the comma expression is a valid null pointer constant, which is also the result of the comma expression, therefore I expect not a reported pointer/integer type mismatch.
